triggers/daml/Daml/Trigger.daml

```diff
@@ -274,10 +274,15 @@ utUpdateState updateState commandsInFlight acs actAs readAs msg =
       let state = TriggerUpdateState commandsInFlight acs actAs readAs
       in execState $ flip runTriggerUpdateA state $ updateState msg
 
-updateImpl : Trigger s -> Bool -> [Message] -> TriggerRule (TriggerState s) ()
-updateImpl _ False [] = pure ()
-updateImpl userTrigger True [] = runRule userTrigger.rule
-updateImpl userTrigger triggered (msg :: rest) = do
+updateImpl : forall s. Trigger s -> [Message] -> TriggerRule (TriggerState s) ()
+updateImpl userTrigger msgs = do
+  runUserTrigger <- or <$> mapA processMessage msgs
+  when runUserTrigger $
+    runRule userTrigger.rule
```

Comment:
I didn't even need `foldl`, a simple `mapA` followed by `or`ing the results was enough

Reply:
@remyhaemmerle-da @carlpulley-da

Reply:
Thanks for helping out here